rescan.go

```diff
@@ -773,12 +792,10 @@ rescanLoop:
 					return fmt.Errorf("unable to register "+
 						"block subscription: %v", err)
 				}
-				defer func() {
-					if blockSubscription != nil {
-						blockSubscription.Cancel()
-						blockSubscription = nil
-					}
-				}()
+
+				// Clear any retry blocks as we are now
+				// considered current with the chain.
+				retryBlocks = nil
```

Comment:
Can there still be retryBlocks at this point?

Reply:
Yes, if we fail to process a notification in `handleBlockConnected` after queueing blocks.